server/src/internalClusterTest/java/org/opensearch/blocks/CreateIndexBlockIT.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.blocks;
+
+import org.junit.After;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import static org.opensearch.test.OpenSearchIntegTestCase.client;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertBlocked;
+
+public class CreateIndexBlockIT extends OpenSearchIntegTestCase {
+
+    public void testBlockCreateIndex() {
+        setCreateIndexBlock("true");
+        assertBlocked(client().admin().indices().prepareCreate("diskguardrails1"), Metadata.CLUSTER_CREATE_INDEX_BLOCK);
+        setCreateIndexBlock("false");
+        assertAcked(client().admin().indices().prepareCreate("diskguardrails2").execute().actionGet());
+    }
+
+    @After
+    public void cleanup() throws Exception {
+        Settings settings = Settings.builder().putNull(Metadata.SETTING_CREATE_INDEX_BLOCK_SETTING.getKey()).build();
+        assertAcked(client().admin().cluster().prepareUpdateSettings().setTransientSettings(settings).get());
```

Comment:
Is this necessary? I don't think this cluster is shared with other test cases so I'd remove this.

Reply:
Without this, we started getting error like: "java.lang.AssertionError: test leaves transient cluster metadata behind"